src/tests/ftest/control/dmg_pool_query_ranks.py

```diff
@@ -0,0 +1,161 @@
+        :avocado: tags=vm
+        :avocado: tags=dmg,pool_query,pool_query_ranks
+        :avocado: tags=pool_query_ranks_error
+        """
+        self.log.info("Tests of pool query with incompatible options")
+
+        # Disable raising an exception if the dmg command fails
+        self.dmg.exit_status_exception = False
+        try:
+            data = self.dmg.pool_query(self.pool.identifier, show_enabled=True, show_disabled=True)
+            self.assertIsNotNone(data["error"], "Invalid error field: want={}, "
+                    "got=None".format(data['error']))
+            self.assertIn(r'may not be mixed with', str(data['error']), "Invalid error message")
+        finally:
+            self.dmg.exit_status_exception = True
+
+    def test_pool_query_ranks_mgmt(self):
+        """Test the state of ranks after excluding and reintegrate them
+
+        Test Description:
+            Create a pool with some engines exclude them one by one and check the consistency of the
+            list of enabled and disabled ranks.  Then, reintegrate them and check the consistency of
+            the list of enabled and disabled ranks.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=dmg,pool_query,pool_query_ranks
+        :avocado: tags=pool_query_ranks_mgmt
+        """
+        self.log.info("Tests of pool query with ranks state when playing with ranks")
+
+        enabled_ranks = [0, 1, 2]
+        disabled_ranks = []
+        for rank in [1, 0, 2]:
+            self.log.debug("Excluding rank %d", rank)
+            self.dmg.pool_exclude(self.pool.uuid, rank)
+            enabled_ranks.remove(rank)
+            disabled_ranks = sorted(disabled_ranks + [rank])
+
+            self.log.debug("Checking enabled ranks state information")
+            data = self.dmg.pool_query(self.pool.identifier, show_enabled=True)
+            self.assertListEqual(data['response']['enabled_ranks'], enabled_ranks,
+                    "Invalid enabled_ranks field: want={}, "
+                    "got={}".format(enabled_ranks, data['response']['enabled_ranks']))
+
+            self.log.debug("Checking disabled ranks state information")
+            data = self.dmg.pool_query(self.pool.identifier, show_disabled=True)
+            self.assertListEqual(data['response']['disabled_ranks'], disabled_ranks,
+                    "Invalid disabled_ranks field: want={}, "
+                    "got={}".format(disabled_ranks, data['response']['disabled_ranks']))
+
+        for rank in [2, 0, 1]:
+            self.log.debug("Reintegrating rank %d", rank)
+            self.dmg.exit_status_exception = False
+            # NOTE Reintegrating ranks could not be done immediately => looping with timeout until
+            # the eviction is done.
+            timeout = 60
+            try:
+                while timeout > 0:
+                    rc = self.dmg.pool_reintegrate(self.pool.uuid, rank)
```

Comment:
```suggestion
                    rc = self.pool.reintegrate(rank=rank)
```
In general, it's better to use the `TestPool` method.

Also, can't we use `TestPool.wait_for_rebuild()` to check whether the reintegrate is done?

Reply:
Did not know the second method, it should greatly simplify the test => thanks for the tips :)